tools/python/src/other.cpp

```diff
@@ -264,5 +278,69 @@ ensures \n\
 
     m.def("probability_that_sequence_is_increasing",probability_that_sequence_is_increasing, py::arg("time_series"),
         "returns the probability that the given sequence of real numbers is increasing in value over time.");
+
+    {
+        typedef momentum_filter type;
+        py::class_<type>(m, "momentum_filter",
+            R"asdf( 
+                This object is a simple tool for filtering a single scalar value that
+                measures the location of a moving object that has some non-trivial
+                momentum.  Importantly, the measurements are noisy and the object can
+                experience sudden unpredictable accelerations.  To accomplish this
+                filtering we use a simple Kalman filter with a state transition model of:
+
+                    position_{i+1} = position_{i} + velocity_{i} 
+                    velocity_{i+1} = velocity_{i} + some_unpredictable_acceleration
+
+                and a measurement model of:
+                    
+                    measured_position_{i} = position_{i} + measurement_noise
+
+                Where some_unpredictable_acceleration and measurement_noise are 0 mean Gaussian 
+                noise sources with standard deviations of get_typical_acceleration() and
+                get_measurement_noise() respectively.
+
+                To allow for really sudden and large but infrequent accelerations, at each
+                step we check if the current measured position deviates from the predicted
+                filtered position by more than get_max_measurement_deviation()*get_measurement_noise() 
+                and if so we adjust the filter's state to keep it within these bounds.
+                This allows the moving object to undergo large unmodeled accelerations, far
+                in excess of what would be suggested by get_typical_acceleration(), without
+                then experiencing a long lag time where the Kalman filter has to "catch
+                up" to the new position.  )asdf"
+        )
+        .def(py::init<double,double,double>(), py::arg("measurement_noise"), py::arg("typical_acceleration"), py::arg("max_measurement_deviation"))
+        .def("measurement_noise",   [](const momentum_filter& a){return a.get_measurement_noise();})
+        .def("typical_acceleration",   [](const momentum_filter& a){return a.get_typical_acceleration();})
+        .def("max_measurement_deviation",   [](const momentum_filter& a){return a.get_max_measurement_deviation();})
+        .def("__call__", [](momentum_filter& f, const double r){return f(r); })
```

Comment:
This kind of thing can just be `.def("foo", &type::foo)`.  I'll adjust that in a separate commit though.